lib/runner/cli/cli.js

```diff
@@ -261,6 +264,37 @@ class CliRunner {
       isBstack: this.test_settings.desiredCapabilities['bstack:options'] !== undefined,
       test_runner: this.test_settings.test_runner ? this.test_settings.test_runner.type : null
     });
+
+    this.setupEventHub();
+  }
+
+  isRegisterEventHandlersCallbackExistsInGlobal() {
+    const {globals} = this.test_settings;
+    const {plugins = []} = this.globals;
+  
+    return Utils.isFunction(globals.registerEventHandlers) || plugins.some(plugin => plugin.globals && Utils.isFunction(plugin.globals.registerEventHandlers));
+  }
+
+  setupEventHub() {
+    if (this.isRegisterEventHandlersCallbackExistsInGlobal()) {
+      NightwatchEventHub.runner = this.testRunner.type;
+      NightwatchEventHub.isAvailable = true;
+      
+      const {globals} = this.test_settings;
+      const {plugins} = this.globals;
+  
+      if (Utils.isFunction(globals.registerEventHandlers)) {
+        globals.registerEventHandlers(NightwatchEventHub);
+      }
+  
+      if (plugins.length > 0) {
+        plugins.forEach((plugin) => {
+          if (plugin.globals && Utils.isFunction(plugin.globals.registerEventHandlers)){
+            plugin.globals.registerEventHandlers(NightwatchEventHub);
+          }
+        });
+      }
```

Comment:
Can we move this to `loaders`?

Reply:
@gravityvi I need help in identifying the suitable loader for moving this code. Also the reasoning will greatly help me to understand the transition.